src/Fmt_ast.ml

```diff
@@ -68,8 +68,60 @@ let update_config c ({txt; loc}, payload) =
       !Location.warning_printer loc Caml.Format.err_formatter w ;
       c
 
-let update_config c l =
-  {c with conf= List.fold ~init:c.conf l ~f:update_config}
+let update_config c l = List.fold ~init:c l ~f:update_config
```

Comment:
The shadowing would probably be clearer if the `update_config` above was defined inside the `update_config` here.